### PR TITLE
stm32f0: GPIO enable IRQ for EXTI0_1

### PR DESCRIPTION
**Pull request by botmayank**
Enabling IRQ in gpio.c for EXTI0_1

### CODE REVIEW

**Comment by A-Paul** [comment]
@botmayank, thanks for contributing.
Can you please be more specific in your PR comment. Are you intending this to resolve a bug, making an enhancement?

File: cpu/stm32f0/periph/gpio.c [hunk 1]
```diff
@@ -102,6 +102,7 @@ int gpio_init_int(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb
 
     /* enable global pin interrupt */
     if (pin_num < 2) {
+        NVIC_EnableIRQ(EXTI0_1_IRQ);
         NVIC_EnableIRQ(EXTI2_3_IRQn + pin_num);
```
Comment (DipSwitch) [inline cpu/stm32f0/periph/gpio.c]: this line should be removed. For the STM32F0x[128]

**Comment by botmayank** [comment]
I'm intending to solve a bug. I'm using a board based on the stm32f091cct6 and was using pins PB0 and PB1 for external interrupts however from what I could see, in the GPIO driver EXTI0_1_IRQ isn't enabled anywhere as per this implementation. I kept getting a hardfault in a gpio interrupt init call without this line. Adding this line fixed the issue for me. @DipSwitch please let me know if I'm missing something obvious.

**Comment by DipSwitch** [comment]
Line 106: `NVIC_EnableIRQ(EXTI2_3_IRQn + pin_num);` may be removed I belief, this because the for the: STM32F0x1, STM32F0x2, STM32F0x8, STM32F030xx and STM32F070xx the vector table looks like:

    irq prio type         name
    5   12   settable   EXTI0_1 EXTI Line[1:0] interrupts 0x0000 0054
    6   13   settable   EXTI2_3 EXTI Line[3:2] interrupts 0x0000 0058
    7   14   settable   EXTI4_15 EXTI Line[15:4] interrupts 0x0000 005C

**Comment by botmayank** [comment]
Ok, so how should I proceed? [*my first PR :) ]

**Comment by DipSwitch** [comment]
Checkout your development branch, remove L106 in the file. Then you can `git commit` and squash your commits by using `git rebase -i HEAD~2` set the last commit to `s` or `squash` and then when it's commited you can `git push --force` to update your PR :)

**Comment by botmayank** [comment]
Done, is that fine?

File: cpu/stm32f0/periph/gpio.c [hunk 2]
```diff
@@ -102,7 +102,7 @@ int gpio_init_int(gpio_t pin, gpio_pp_t pullup, gpio_flank_t flank, gpio_cb_t cb
 
     /* enable global pin interrupt */
     if (pin_num < 2) {
-        NVIC_EnableIRQ(EXTI2_3_IRQn + pin_num);
+        NVIC_EnableIRQ(EXTI0_1_IRQ);
```
Comment (DipSwitch) [inline cpu/stm32f0/periph/gpio.c]: Please remove trailing white spaces and I think it should be `EXTI0_1_IRQn` instead of `EXTI0_1_IRQ`

**Comment by DipSwitch** [comment]
Yes! Almost :)

**Comment by botmayank** [comment]
Oops..didn't see the 'n' ..I've removed the white spaces and edited line 105 as you said but how should I continue now? rebase with a continue flag? abort the previous rebase and redo it?

**Comment by PeterKietzmann** [comment]
You should squash your second commit and rename the first commit slightly to "cpu/stm32f0:....."

**Comment by botmayank** [comment]
Squashed, @PeterKietzmann is this fine?

**Comment by PeterKietzmann** [comment]
@botmayank I assume the first commit is not intended (`tests: added SAUL test application`)?

**Comment by botmayank** [comment]
Yes -_- ...how should I remove that?

**Comment by PeterKietzmann** [comment]
When rebasing your branch use `git rebase -i master` and uncomment the unwanted commit (with `#` )

**Comment by botmayank** [comment]
Ok now?

**Comment by PeterKietzmann** [comment]
yep

**Comment by botmayank** [comment]
Cool :D

**Comment by PeterKietzmann** [comment]
All tests passed. ACK and go

**Pull request merged**